traitsui/tests/editors/test_enum_editor.py

```diff
@@ -101,18 +94,237 @@ def set_combobox_index(combobox, idx):
         combobox.setCurrentIndex(idx)
 
 
-class TestEnumEditor(unittest.TestCase):
-    def check_enum_text_update(self, view):
+def finish_combobox_text_entry(combobox):
+    """ Finish text entry given combobox. """
+    if is_current_backend_wx():
+        import wx
+
+        event = wx.CommandEvent(wx.EVT_TEXT_ENTER.typeId, combobox.GetId())
+        wx.PostEvent(combobox, event)
+
+    elif is_current_backend_qt4():
+        combobox.lineEdit().editingFinished.emit()
+
+
+def get_all_button_status(widget):
+    """ Gets status of all radio buttons in the layout of a given widget."""
+    if is_current_backend_wx():
+        button_status = []
+        for item in widget.GetSizer().GetChildren():
+            button = item.GetWindow()
+            if button.value != "":  # There are empty invisible buttons
+                button_status.append(button.GetValue())
+        return button_status
+
+    elif is_current_backend_qt4():
+        layout = widget.layout()
+        button_status = []
+        for i in range(layout.count()):
+            button_status.append(layout.itemAt(i).widget().isChecked())
+        return button_status
+
+
+def click_radio_button(widget, button_idx):
+    """ Simulate a radio button click given widget and button number."""
+    if is_current_backend_wx():
+        import wx
+
+        sizer_items = widget.GetSizer().GetChildren()
+        button = sizer_items[button_idx].GetWindow()
+        event = wx.CommandEvent(wx.EVT_RADIOBUTTON.typeId, button.GetId())
+        event.SetEventObject(button)
+        wx.PostEvent(widget, event)
+
+    elif is_current_backend_qt4():
+        widget.layout().itemAt(button_idx).widget().click()
+
+
+def get_list_widget_text(list_widget):
+    """ Return the text of currently selected item in given list widget. """
+    if is_current_backend_wx():
+        selected_item_idx = list_widget.GetSelection()
+        return list_widget.GetString(selected_item_idx)
+
+    elif is_current_backend_qt4():
+        return list_widget.currentItem().text()
+
+
+def set_list_widget_selected_index(list_widget, idx):
+    """ Set the choice index given a list widget control and index number. """
+    if is_current_backend_wx():
+        import wx
+
+        list_widget.SetSelection(idx)
+        event = wx.CommandEvent(wx.EVT_LISTBOX.typeId, list_widget.GetId())
+        wx.PostEvent(list_widget, event)
+
+    elif is_current_backend_qt4():
+        list_widget.setCurrentRow(idx)
+
+
+class TestEnumEditorMapping(unittest.TestCase):
```

Comment:
Since all the tests are decorated with `skip_if_not_null`, we could also decorate the whole test case instead of decorating individual test methods.

Reply:
Ah just realized my comment wrongly stated `skip_if_not_null`. `skip_if_null` is the correct one. Thanks!